src/main/java/com/kodilla/ecommercee/controller/UserController.java

```diff
@@ -1,27 +1,40 @@
 package com.kodilla.ecommercee.controller;
 
 import com.kodilla.ecommercee.domain.dto.UserDto;
-import org.springframework.web.bind.annotation.*;
-
-import java.util.Random;
+import com.kodilla.ecommercee.exception.UserNotFoundException;
+import com.kodilla.ecommercee.mapper.UserMapper;
+import com.kodilla.ecommercee.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
 
 @RestController
 @RequestMapping("/users")
 public class UserController {
-    @PostMapping("addUser")
-    public UserDto addUser(@RequestBody UserDto userDto) {
-        return userDto;
+    private UserMapper userMapper;
+    private UserService userService;
+
+    @Autowired
+    public UserController(UserMapper userMapper, UserService userService) {
+        this.userMapper = userMapper;
+        this.userService = userService;
     }
 
-    @PutMapping("banUser")
-    public UserDto banUser(@RequestBody UserDto userDto) {
-        return new UserDto(userDto.getUsername(), "-1", userDto.getUserKey());
+    @PostMapping("addUser")
+    public void addUser(@RequestBody UserDto userDto) {
+        userService.addUser(userMapper.mapToUser(userDto));
     }
 
-    @GetMapping("generateKey")
-    public Long generateKey(@RequestParam long userId) {
-        long key = new Random().nextLong();
-        return key + userId;
+    @PatchMapping("banUser")
+    public UserDto banUser(@RequestBody UserDto userDto) throws UserNotFoundException {
```

Comment:
"zablokowanie użytkownika PATCH /users/1" zrezygnował bym też z throws.

Reply:
Poszukaj sobie informacji o `@PathMapping` i sprawdź jak zrobił to @aleksanderkot00, `@PathMapping` przyjmuje `id` i `@RequestBody` wtedy service aktualizuje pola które są do zaktualizowania.

Reply:
ok, wiem o tym, z tym że jeśli dobrze rozumiem banowanie użytkownika to tylko zmiana statusu użytkownika na "banned" dlatego nie potrzebuję body a jedynie id użytkownika do zbanowania

Reply:
testowałeś to postmanem?

Reply:
tak, działa ok u mnie

Reply:
wydaje mi się że powinieneś zrobić tak jak @aleksanderkot00 tylko że powinieneś umożliwić aktualizację jedynie statusu.

Reply:
Początkowo miałem tak zrobione ale potem tego body nigdzie nie używałem więc usunąłem

Reply:
u mnie niestety nie działa, może robię coś źle https://drive.google.com/drive/folders/1rtT40ZSq2DlGY9cZcUNDmoy2bGTO9DCs?usp=sharing sprawdź proszę i napisz mi co o tym myślisz

Reply:
jak byś zrobił jak pisałem i zmienił nazwę metody to mogła by ona służyć nie tylko do banowania

Reply:
Poprawny link do testowania banUser to http://localhost:8080/users/id  Twoja wersja byłaby poprawna gdybym miał @RequestParam

Reply:
faktycznie :) zmęczony już jestem idę spać, a ty pomyśl czy bana nie powinno się dać cofnąć.